runtime-manager/src/managers/mod.rs

```diff
@@ -184,14 +184,13 @@ impl ProtocolState {
     }
 
     /// Requests shutdown on behalf of a client, as identified by their client
-    /// ID, and then checks if this request was sufficient to reach a threshold
-    /// of requests wherein the platform can finally shutdown.
+    /// ID.
+    /// TODO: Do something better (https://github.com/veracruz-project/veracruz/issues/393)
     pub(crate) fn request_and_check_shutdown(
         &mut self,
-        client_id: u64,
+        _client_id: u64,
     ) -> Result<bool, RuntimeManagerError> {
-        self.expected_shutdown_sources.retain(|v| v != &client_id);
-        Ok(self.expected_shutdown_sources.is_empty())
+        Ok(!self.expected_shutdown_sources.is_empty())
```

Comment:
The boolean here is not as expected.
- Directly return `Ok(true)`  here? We basically by-pass `expected_shutdown_sources`? 
- Or a better way is to check if a `client_id` is in the `expected_shutdown_sources` and then return `Ok(true)`.